### PR TITLE
[stable/mail2most] added mail2most chart

### PR DESCRIPTION
**Pull request by cseeger-epages**
Signed-off-by: Carsten Seeger <c.seeger@epages.com>

#### What this PR does / why we need it:

This PR adds a new chart for [mail2most](https://github.com/cseeger-epages/mail2most) a small tool to parse mails, filter them and send them to mattermost

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: incubator/mail2most/templates/pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{{ if .Values.persistence.enabled -}}
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ template "mail2most.fullname" . }}-pvc
+  labels:
+    app: {{ template "mail2most.fullname" . }}
```
Comment (cpanato) [inline incubator/mail2most/templates/pvc.yaml]: use the new labels, see: https://github.com/helm/helm/blob/master/docs/chart_best_practices/labels.md#standard-labels

File: incubator/mail2most/values.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+deployment:
+  replicas: 1
+  image:
+    repository: virtomize/mail2most
+    tag: latest
+    pullPolicy: Always
+
+resources:
```
Comment (cpanato) [inline incubator/mail2most/values.yaml]: resources should be {}

File: incubator/mail2most/values.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  limits:
+    cpu: 200m
+    memory: 256Mi
+  requests:
+    cpu: 100m
+    memory: 128Mi
+
+persistence:
+  enabled: false
```
Comment (cpanato) [inline incubator/mail2most/values.yaml]: enabled by default

File: incubator/mail2most/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+description: A Helm chart for Mail2Most a tool for parsing emails to be send to mattermost
+name: mail2most
+version: 1.1.0
+appVersion: "1.0.0"
```
Comment (cpanato) [inline incubator/mail2most/Chart.yaml]: can remove the ""

**Review by cpanato — CHANGES_REQUESTED** [review]

File: incubator/mail2most/Chart.yaml (hunk 4 above)
Comment (cseeger-epages) [reply incubator/mail2most/Chart.yaml]: oh i see the app version is 1.1.0 and the chart version is 1.0.0 :D ill fix this

**Comment by cseeger-epages** [comment]
@cpanato, @jlegrone is there anything that needs to be changed ?

File: incubator/mail2most/templates/configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
```
Comment (cpanato) [inline incubator/mail2most/templates/configmap.yaml]: add the labels here as well

File: incubator/mail2most/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app:  {{ template "name" . }}
```
Comment (cpanato) [inline incubator/mail2most/templates/deployment.yaml]: please use the new labels see: https://github.com/helm/helm/blob/master/docs/chart_best_practices/labels.md#standard-labels

**Review by cpanato — CHANGES_REQUESTED** [review]
you can move this chart to the stable folder as well

**Comment by cseeger-epages** [comment]
@cpanato , @jlegrone more suggestions here ?

**Comment by cseeger-epages** [comment]
@cpanato , @jlegrone , @prydonius  is there anything that i can do to help ?

**Comment by maorfr** [comment]
/ok-to-test

cc @cpanato

File: stable/mail2most/templates/configmap.yaml [hunk 7]
```diff
@@ -0,0 +1,61 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "name" . }}-configmap
```
Comment (cpanato) [inline stable/mail2most/templates/configmap.yaml:4]: please use the labels here as well

File: stable/mail2most/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,37 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "name" . }}
+  labels:
+    app:  {{ template "name" . }}
```
Comment (cpanato) [inline stable/mail2most/templates/deployment.yaml]: please update to use the new labels, see: https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#metadata

**Review by cpanato — APPROVED** [review]
see my comments

File: stable/mail2most/values.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
+deployment:
+  replicas: 1
+  image:
+    repository: virtomize/mail2most
+    tag: latest
```
Comment (cpanato) [inline stable/mail2most/values.yaml]: please dont use latest, does this image have a release or a tag version? if not consider to make it

File: stable/mail2most/values.yaml [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    pullPolicy: Always
```
Comment (cpanato) [inline stable/mail2most/values.yaml]: move to `IfNotPresent` when have the tag

File: stable/mail2most/values.yaml [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+resources: {}
+#  limits:
+#    cpu: 200m
+#    memory: 256Mi
+#  requests:
+#    cpu: 100m
+#    memory: 128Mi
+
+persistence:
+  enabled: true
```
Comment (cpanato) [inline stable/mail2most/values.yaml:17]: please add the other options for the persistence as well see this chart for an example: https://github.com/helm/charts/blob/master/stable/chartmuseum/values.yaml#L179 or any other chart in the stable folder

Comment (cseeger-epages) [reply stable/mail2most/values.yaml:17]: @cpanato can you take a look if this is correct ?

**Comment by cseeger-epages** [comment]
@cpanato can you give me a hint what to do to make the `pull-charts-e2e ` work ? For me it looks like an internal script error.

**Comment by cseeger-epages** [comment]
@cpanato , @jlegrone , @prydonius anything i missed here ?

File: stable/mail2most/templates/pvc.yaml [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+metadata:
+  name: {{ template "name" . }}-pvc
+  labels:
+    app.kubernetes.io/name:  {{ template "name" . }}
+    app.kubernetes.io/instance: "{{ .Release.Name }}"
+    app.kubernetes.io/managed-by: "{{ .Release.Service }}"
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+spec:
+  accessModes:
+    - ReadWriteMany
```
Comment (cpanato) [inline stable/mail2most/templates/pvc.yaml]: in the values you define `ReadWriteOnce` and here is `ReadWriteMany` make it configurable

File: stable/mail2most/values.yaml [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+  ## Mail2Most data Persistent Volume Storage Class
+  ## If defined, storageClassName: <storageClass>
+  ## If set to "-", storageClassName: "", which disables dynamic provisioning
+  ## If undefined (the default) or set to null, no storageClassName spec is
+  ##   set, choosing the default provisioner.  (gp2 on AWS, standard on
+  ##   GKE, AWS & OpenStack)
+  ##
+  # storageClass: "-"
+  # volumeName:
+  pv:
```
Comment (cpanato) [inline stable/mail2most/values.yaml:34]: where this is being used?

**Review by cpanato — CHANGES_REQUESTED** [review]

File: stable/mail2most/values.yaml (hunk 13 above)
Comment (cseeger-epages) [reply stable/mail2most/values.yaml:34]: added the missing `templates/pv.yaml`

**Comment by cseeger-epages** [comment]
@cpanato anything left to do ?

**Comment by cseeger-epages** [comment]
@cpanato ping

File: stable/mail2most/templates/deployment.yaml [hunk 14]
```diff
@@ -0,0 +1,37 @@
+apiVersion: extensions/v1beta1
```
Comment (cpanato) [inline stable/mail2most/templates/deployment.yaml:1]: update to `apps/v1` extensions is deprecated on 1.16

**Review by cpanato — CHANGES_REQUESTED** [review]

**Comment by scottrigby** [comment]
Hi 👋 new charts are no longer accepted to `stable` or `incubator`. See [Status of the Project](https://github.com/helm/charts#status-of-the-project) and [Deprecation Timeline](https://github.com/helm/charts#deprecation-timeline) for more info about that.

Good news is you can host a Helm chart yourself on GitHub, use the same tools as this repo does for automatic Chart [testing](https://github.com/helm/chart-testing) and [releasing](https://github.com/helm/chart-releaser), and list your Helm repo on https://helm.hub.sh. Follow this issue for info about existing guidelines and upcoming improvements: https://github.com/helm/charts/issues/7800

**Comment by scottrigby** [comment]
@cseeger-epages it's been a while, but you put a lot of work into this chart. Would love to see it up on the Hub. Ping us on Slack if you need any help with setting that up

**Comment by cpanato** [comment]
@cseeger-epages you can copy this pr and open in the mattermost-helm repository. We can host that if you would like

**Comment by cseeger-epages** [comment]
hi guys sorry for the long time without anything going on here. Ill take a look at this as soon as i have time. The chart will definetly be ready, maybe i will rework this for helmv3 since i use the beta for quite some time. @cpanato ill take a look where to host this. Thanks for your help guys you helped me a lot to understand helm. :+1:

**Pull request closed without merging**